src/storage/mod.rs

```diff
@@ -109,6 +109,7 @@ where
 pub unsafe trait DistinctStorage {}
 
 /// The status of an `insert()`ion into a storage.
+#[must_use]
```

Comment:
Actually, couldn't this simply be a `Result<Option<T>>`?

Reply:
I think I wanted to do this before but decided against it because of breakage.

Reply:
My suggestion: Use the error from `error.rs` (create one if there is no such variant yet) and make this simply a type def.

Reply:
Can you clarify what that type def would look like?  I'm a bit confused.

Reply:
Sorry, I shouldn't be comenting here given my limited time.

```
pub type InsertResult<T> = Result<Option<T>, MyError>;
```